web/src/app/core/services/token.service.ts

```diff
@@ -0,0 +1,106 @@
+import { Injectable } from '@angular/core';
+import { throwError, Observable } from 'rxjs';
+import { first, map, switchMap } from 'rxjs/operators';
+import { v4 as uuid } from 'uuid';
+
+// Dashboard model and services
+import { IProjectTokenModel, ProjectModel } from '../../shared/models/index.model';
+import { ProjectService } from './project.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TokenService {
+
+  constructor(
+    private projectService: ProjectService
+  ) { }
+
+  // This function returns the token details via id
+  public findOneById(projectUid: string, tokenUid: string): Observable<IProjectTokenModel> {
+    return this.projectService.findOneById(projectUid)
+      .pipe(
+        map((data: ProjectModel) => data && data.tokens && data.tokens.find((item: IProjectTokenModel) => item.uid === tokenUid))
+      );
+  }
+
+  // This function returns the tokens list
+  public findProjectTokens(projectUid: string): Observable<IProjectTokenModel[]> {
+    return this.projectService.findOneById(projectUid)
+      .pipe(
+        map((data: ProjectModel) => Array.isArray(data.tokens) ? data.tokens : [])
+      );
+  }
+  // This function returns the tokens via name
+  public findProjectTokenByName(projectUid: string, name: string): Observable<IProjectTokenModel[]> {
+    return this.findProjectTokens(projectUid)
+      .pipe(
+        map((tokens: IProjectTokenModel[]) => tokens.filter((item: IProjectTokenModel) => item.name === name))
+      );
+  }
+
+  // This function create the project token
+  public create(projectUid: string, data: IProjectTokenModel): Observable<IProjectTokenModel> {
+    let token: IProjectTokenModel = {
+      uid: uuid(),
+      ...data,
```

Comment:
Thinking about it, it would be good to add some dates:

```typescript
createdOn: firebase.firestore.Timestamp.fromDate(new Date()),
updatedOn: firebase.firestore.Timestamp.fromDate(new Date()),
```